secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,244 @@
+//
+//  RecoveryPhraseValidation.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import Foundation
+import ComposableArchitecture
+import SwiftUI
+
+typealias RecoveryPhraseValidationStore = Store<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+typealias RecoveryPhraseValidationViewStore = ViewStore<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+
+/// Represents the fulfillment (completion) of a group of recovery words by the addition of one word into the given group
+struct RecoveryPhraseStepFulfillment: Equatable {
```

Comment:
This `RecoveryPhraseStepFulfillment` is the data that represents the action of completing a spot of the puzzle with a given word.

Reply:
It's a specific word a specific index and you're using an array of them to perform validation correct?  I don't think this type represents an action at all, it's just an index and a string.  Because it is used as a collection, I think `ValidationWord` is much more appropriate in this case, unless I'm missing something.

Reply:
It represents a _past_ action of the user. I don't like the name either, but `ValidationWord` doesn't easily represent what that struct is about though

Reply:
It can represent the outcome of a past action, but it doesn't represent the action itself imho.  Another way to think about that is that the unit tests don't have actions, you can unit test the validation and data management here without performing any actions, and you would do that with these `ValidationWord`s.

Reply:
Since this is a Word used for Validation, and that's all the data that it holds, what part of it do you think we are not representing with the name "ValidationWord"?

Reply:
Another follow up question for thought, what if the UI changed and there wasn't drag and drop anymore, but instead the user picked between completed versions of a phrase?  In this scenario, there would still be an incomplete phrase but various words placed at various indexes would be used for validation, and this struct would still represent that data.  Should the UI requirements shape the naming of our data model?